dev/Interop/CameraCaptureUI/CameraCaptureUI/CameraCaptureUI.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#include "pch.h"
+#include <future>
+#include <ppltasks.h>
+#include <pplawait.h>
+#include <ShlObj.h>
+#include "CameraCaptureUI.h"
+#include "Microsoft.Windows.Media.Capture.CameraCaptureUI.g.cpp"
+#include "CameraCaptureUITelemetry.h"
+#include "TelemetryHelper.h"
+
+using namespace winrt::Windows::UI;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::ApplicationModel::DataTransfer;
+
+static constexpr const wchar_t* CameraAppPackageName = L"Microsoft.WindowsCamera_8wekyb3d8bbwe";
+static constexpr const wchar_t* CameraPickerUri = L"microsoft.windows.camera.picker:";
+
+
+namespace winrt::Microsoft::Windows::Media::Capture::implementation
+{
+    IAsyncOperation<hstring> LaunchCameraForResultToken(winrt::Microsoft::UI::WindowId const& targetWindow, ValueSet const& properties)
+    {
+        LauncherOptions options;
+        if (targetWindow.Value)
+        {
+            check_hresult(options.as<IInitializeWithWindow>()->Initialize(reinterpret_cast<HWND>(targetWindow.Value)));
+        }
+
+        options.TreatAsUntrusted(false);
+        options.DisplayApplicationPicker(false);
+        options.TargetApplicationPackageFamilyName(CameraAppPackageName);
+
+        Uri launchUri(CameraPickerUri);
+        winrt::Windows::System::LaunchUriResult result = co_await Launcher::LaunchUriForResultsAsync(launchUri, options, properties);
+        if (!result.Result())
+        {
+            co_return L"";
+        }
+        else if (auto t = result.Result().TryLookup(L"SelectedTokens"))
+        {
+            co_return t.as<hstring>();
+        }
+        else
+        {
+            throw hresult_canceled();
```

Comment:
What is hresult_canceled() and why we need to throw it?

Reply:
It's simply an H_result error. It's not meant to occur; the purpose is to indicate that this situation shouldn't happen. We needed to throw an error, so we chose hresult_canceled to signify that something went wrong.

In my testing, even if I cancel the camera capture operation or force-terminate it from the task manager during capture, it returns a null result that the user can handle. If we encounter hresult_canceled, it indicates a significant and unexpected issue that ideally shouldn't occur.

Reply:
It is ERROR_CANCELLED HRESULT error code - https://learn.microsoft.com/en-us/uwp/cpp-ref-for-winrt/error-handling/hresult-error.

For HRESULT error code - https://learn.microsoft.com/en-us/uwp/cpp-ref-for-winrt/error-handling/hresult-error.

Why we need to throw hresult_canceled vs. hresult_error? Should we include `message` value so that developers understand what is the issue?